File: Codility/lessons/BinaryGap.js
/**
 * Lesson 1. Iterations
 */
function solution(N) {
    const str = N.toString(2);
    let max = 0;
    let cnt = 0;
    for (let i = 0; i < str.length; i++) {
        if (str.charAt(i) == "1") {
            if (max < cnt) {
                max = cnt;
            }
            cnt = 0;
        } else {
            cnt++;
        }
    }
    return max;
}

File: Codility/src/lesson1/BinaryGap.java
package lesson1;

public class BinaryGap {
    public int solution(int N) {
        String binaryString = Integer.toBinaryString(N);
        int maxLength = 0;
        int startIndex = -1;

        for (int i = 0; i < binaryString.length(); i++) {
            if (binaryString.charAt(i) == '1') {
                if (startIndex != -1) {
                    maxLength = Math.max(maxLength, i - startIndex - 1);
                }
                startIndex = i;
            }
        }

        return maxLength;
    }
}